draft-ietf-rats-corim.md

```diff
@@ -1840,6 +1832,26 @@ Regardless of the specific integrity protection method used, the Evidence's inte
 > * How key material is associated with the Attesting Environment
 > * How the Attesting Environment is identified in Evidence
 
+##### Example: DICE
+In DICE, a proof of liveness is carried out on the final key in the certificate chain (a.k.a., the alias certificate).
+If this is successful, a suitable certification path is looked up in the Appraisal Context, based on linking information obtained from the DeviceID certificate.
+See Section 9.2.1 of {{DICE.Layer}}.
+If a trusted root certificate is found, the usual X.509 certificate validation is performed.
+
+##### Example: ARM PSA Token
+In PSA {{-psa-token}} the verification public key is looked up in the appraisal context using the `ueid` claim found in the PSA claims-set.
+If found, COSE Sign1 verification is performed accordingly.
+
+##### Example: `attest-key-triple`
+
+A more general method of verifying evidence purely with CoRIM constructs is by verifying the signature with a trusted attestation key associated with the Evidence's Target Environment.
+The Verifier MAY use environment-identifying elements of unverified Evidence to determine the Target Environment.
+The Verifier MAY establish trust in an attestation key by constructing a certificate path to an accepted trust anchor whose constraints apply to the Target Environment.
+For example, the Verifier has accepted a CoTS tag {{-ta-store}} with an environment that matches the `attest-key-triple`'s `environment-map` and a listed purpose is `"eat"`.
+The Verifier MAY associate an attestation key with the Target Environment if a valid `attest-key-triple`'s `environment-map` successfully compares with the Evidence's Target Environment.
```

Comment:
Given this topic is still relatively fresh, I understand the reluctance to add strong statements.  However, these MAYs totally lack bite.  I think a better way to phrase this is to state clear requirements under a set of well-defined circumstances.  For example:

> If `attest-key-triple`s are used during phase 1 for evidence verification, then:
> * The verifier MUST ...
> * The verifier SHOULD ...

Reply:
Okay, if I use this method in the SEV-SNP CoRIM profile, then I can add that normative language for the profile, but this is still an example, and that profile doesn't _actually_ use CoRIMs in this way.

For that profile I will still be making suggestions that Verifiers include a CoTS tag binding the SEV-SNP attestation class-id for the AMD-published keys as an implicit input, and then treating VCEK certificates from KDS as "virtual" attest-key-triples, since they don't issue CoRIMs.

When I describe translating an ATTESTATION_REPORT to claim structure that lines up with reference values in the CoRIM profile, I still describe the whole verification scheme.

It all just feels more like a mental exercise than an actual use of the CoRIM representation.

Reply:
We seem to have similar requirements for PSA/CCA.  Maybe we could sit down and see if we can extract a core set of requirements that are profile-independent.

Reply:
The verifier will accept Evidence assertions so long as the channel to the Attester is authenticated (but not attested). Attest-key triple is not second guessing this assumption.

Phase 3 (corroboration of ref values) may be a better phase to process attest-key triples since the enforcement of a key measurement that fails key verification would reasonably result in that measurement NOT being corroborated. 

It might make sense to require the RIM author to include RV triples that do binary compare operations as a pre-requisite to applying attest-key processing.

Reply:
RV triples that do binary compare operations of what? I'm not following.